service-tests/json.js

```diff
@@ -0,0 +1,44 @@
+'use strict';
+
+const Joi = require('joi');
+const ServiceTester = require('./runner/service-tester');
+
+const t = new ServiceTester({ id: 'badge/dynamic/json', title: 'User Defined JSON Source Data' });
+module.exports = t;
+
+t.create('Connection error')
+  .get('.json?url=https://github.com/badges/shields/raw/master/package.json&query=$.name&label=Package Name')
```

Comment:
I have a minor stylistic preference to call this parameter `uri` instead of `url`.

Reply:
Haha yeah, i wasn't sure whether to put `uri` or `url` as the param, but i guess you don't **need** to specify the protocol in the param so will update this to be `uri` unless we do want the protocol to always be specified?